addons/mail/static/src/core/common/message_model.js

```diff
@@ -187,7 +187,11 @@ export class Message extends Record {
      * @returns {boolean}
      */
     get allowsEdition() {
-        return this.store.self_user?.is_admin || this.isSelfAuthored;
+        return (
+            this.store.self_user?.is_admin ||
+            this.isSelfAuthored ||
+            ["owner", "admin"].includes(this.thread?.channel?.self_member_id?.channel_role)
```

Comment:
Note: as this condition is purely channel related, it should be an override in discuss folder instead